samples/terminate/src/main.c

```diff
@@ -0,0 +1,182 @@
+
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <string.h>
+#include <unistd.h>
+#include <pthread.h>
+
+#include "wasm_export.h"
+#include "bh_read_file.h"
+#include "bh_getopt.h"
+
+void
+print_usage(void)
+{
+    fprintf(stdout, "Options:\r\n");
+    fprintf(stdout, "  -f [path of wasm file] \n");
+}
+
+static void *
+runner(void *vp)
+{
+    wasm_module_inst_t inst = vp;
+    bool ok = wasm_runtime_init_thread_env();
+    assert(ok);
+    wasm_application_execute_main(inst, 0, NULL);
+    return inst;
```

Comment:
Had better call `wasm_runtime_destroy_thread_env` before return?

Reply:
sure. i added it.